### ISSUE TITLE
Dynamic Asset Loading Issue with Enabled Cache in Modular Templates

### ISSUE THREAD

**Issue by tiggeymone** [getgrav/grav#3811]
I've encountered a problem regarding dynamic asset loading in a modular template within a Grav theme. Specifically, the issue arises when attempting to dynamically add CSS assets to a modular template. The expected behavior is for these assets to be loaded into the <head> of the page whenever the module is used.

```
{% set body_classes = ['my-theme-heroic-quote', page.header.body_classes ? page.header.body_classes : '']|join(' ')|trim %}
{% do assets.addCss('theme://css/components/heroic_quote/heroic-quote.css') %}
{% do assets.addCss('theme://css/components/heroic_quote/heroic-quote-custom.css') %}
<div class="{{ body_classes }}">
    <blockquote>{{ page.header.citation|raw }}</blockquote>
    <div class="quote-author">{{ page.header.author|raw }}</div>
</div>
```

**Observed Behavior:**

With caching disabled, the CSS files are correctly loaded in the <head> of the page, and everything works as expected.
With caching enabled, the CSS files load correctly on the first render. However, on subsequent page loads, the CSS files are no longer injected into the <head> of the page.

**Questions/Issues:**

- Is this behavior expected when the cache is enabled, and dynamic assets are added through modular templates?
- Is there a recommended approach to ensure that dynamic assets are loaded only when a module is actually invoked on a page, especially in the context of enabled caching?

**Environment**:

Grav version: 1.7.44

I've reviewed the documentation and community forums but haven't found a clear solution to this specific issue. Any guidance on how to properly manage dynamic asset loading in modular templates with caching would be greatly appreciated.

Thank you for your assistance.

**Comment by rhukster** [comment]
Just set `never_cache_twig: true` in the frontmatter.

**Comment by tiggeymone** [comment]
Thank you @rhukster , incredibly, after hours of research, I found the related discussion just a second before your reply: https://github.com/getgrav/grav/issues/1934

Ok, now the mechanism is clear to me. Is there a way to make this the default behavior for certain modules? I tried to override the blueprint:

```
form:
  fields:
    tabs:
      type: tabs
      active: 1
      fields:
...
        advanced:
          fields:
            overrides:
              fields:
                header.never_cache_twig:
                  default: true
```
But the change only occurs in the admin panel.

One last thing: by disabling "Twig Caching" in the system options, shouldn't I achieve the same result? Because 'never_cache_twig: true' works, but "Twig caching: No" does not.

Thanks again and kudos for the work you do!
